Add authorizer parsing to proxy lambda request context

wuzzeb opened the pull request:
This allows access to the authorizer propery on the api gateway lambda request context.  The authorizier propery is used during cognito user pools where authorizer.claims contains the claims from the token.  The authorizer is also used with lambda authorizers where the return value from the lambda authorizer claims is put directly on the authorizer property.

Documentation is here: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html

dmoverton commented:
Could you please fix the tests so that they build and pass.

dmoverton commented on src/AWSLambda/Events/APIGateway.hs:
```diff
@@ -94,6 +94,19 @@ instance FromJSON RequestIdentity where
 
+data Authorizer = Authorizer
+  { _aPrincipalId :: !Text
+  , _aClaims :: !(HashMap Text Text)
```
I don't know much about Cognito, but are the claims values always guaranteed to be strings or could they be other JSON values? If so, this should be an `Object` rather than `HashMap Text Text`.

wuzzeb commented:
I looked at the docs closer and you were right, the claims can be anything so I switched it to Object... my own testing just had only text.

I also updated the test suite.

dmoverton commented:
Thanks. Looks good.
